### DISCUSSION TITLE
CalculateRollupFieldRequest workaround

### DISCUSSION THREAD

**Discussion by bwmodular** [jordimontana82/fake-xrm-easy#536] [Q&A] [answered]
I know that this request isn't implemented in fakeXrmEasy, and I can see why not - it would have to be able to retrieve the definition of the rollup field from the metadata and then perform the rollup before returning the result.

However, I'm wondering if I could implemenent a local mock which would at least allow the response to be called and return a mocked result. I'm writing tests for a plugin which calls CalculateRollupFieldRequest - it doesn't really matter in the test what the result is, it just calls it so the field is updated when the plugin completes its work. At the moment my tests are failing because  CalculateRollupFieldRequest isn't implemented - but as my plugin doesn't care what the result is, I wonder if I can get round this somehow.

I could of course just use a local copy of FakeXrmEasy which mocks the request, but that makes ongoing maintenance harder.

Thanks as ever for any ideas...

**Comment by BetimBeja** [comment, marked as answer]
Hello @bwmodular , you can use [AddGenericFakeMessageExecutor](https://github.com/jordimontana82/fake-xrm-easy/blob/f294b9af7319f8e23d4032a9290ec75a628f2435/FakeXrmEasy.Shared/XrmFakedContext.cs#L210-L216) to register the custom CalculateRollupFieldRequestExecutor

**Reply by bwmodular** [reply]
Awesome! Thanks @BetimBeja

**Reply by jordimontana82** [reply]
You have 3 options to override the default implementation of messages or messages that are not yet implemented. 

- One option is @DigitalFlow's  [AddExecutionMock](https://github.com/jordimontana82/fake-xrm-easy/blob/5531248d7e9dc2a112b17735e0bf5939e974d342/FakeXrmEasy.Shared/XrmFakedContext.cs#L184). If you just want to override something ocasionally, AddExecutionMock might be the easiest since you can override one fake message easily inline with your unit test. 

- If you want to override a message that is meant to be more generic and reused across tests, then the [AddFakeMessageExecutor](https://github.com/jordimontana82/fake-xrm-easy/blob/f294b9af7319f8e23d4032a9290ec75a628f2435/FakeXrmEasy.Shared/XrmFakedContext.cs#L197) is meant to be used for that, since you can encapsulate the logic in a separate class and pass it to your context in every test 

- Finally, the option suggested above [AddGenericFakeMessageExecutor](https://github.com/jordimontana82/fake-xrm-easy/blob/f294b9af7319f8e23d4032a9290ec75a628f2435/FakeXrmEasy.Shared/XrmFakedContext.cs#L210-L216) is meant to be used for mocking OrganizationRequests that don't have types generated for these messages (i.e. late bound custom actions, or even out of the box requests that don't exist in the SDK).

**Reply by bwmodular** [reply]
Thanks both - the AddExecutionMock is exactly what I need for my purposes - simple enough that even I could understand how to use it ;-)
